kentikapi/users_integration_test.go

```diff
@@ -598,6 +602,366 @@ func TestClient_DeleteUser(t *testing.T) {
+				Role:         "Member",
+				EmailService: true,
+				EmailProduct: false,
+			}),
+			expectedRequestBody: object{
+				"user": object{
+					"username":       "testuser",
+					"user_full_name": "Test User",
+					"user_email":     "test@user.example",
+					"role":           "Member",
+					"email_service":  true,
+					"email_product":  false,
+				},
+			},
+			responses: []httpResponse{
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				{http.StatusCreated,
+					`{
+						"user": {
+							"id": "145999",
+							"username": "testuser",
+							"user_full_name": "Test User",
+							"user_email": "test@user.example",
+							"role": "Member",
+							"email_service": "true",
+							"email_product": "false",
+							"last_login": null,
+							"created_date": "2020-12-09T14:48:42.187Z",
+							"updated_date": "2020-12-09T14:48:43.243Z",
+							"company_id": "74333",
+							"user_api_token": null,
+							"filters": {},
+							"saved_filters": []
+						}
+					}`,
+				},
+			},
+			expectedResult: &models.User{
+				ID:           145999,
+				Username:     "testuser",
+				UserFullName: "Test User",
+				UserEmail:    "test@user.example",
+				Role:         "Member",
+				EmailService: true,
+				EmailProduct: false,
+				LastLogin:    nil,
+				CreatedDate:  *testutil.ParseISO8601Timestamp(t, "2020-12-09T14:48:42.187Z"),
+				UpdatedDate:  *testutil.ParseISO8601Timestamp(t, "2020-12-09T14:48:43.243Z"),
+				CompanyID:    74333,
+				UserAPIToken: nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// arrange
+			h := newSpyHTTPHandler(t, tt.responses)
+			s := httptest.NewServer(h)
+			defer s.Close()
+
+			c := kentikapi.NewClient(kentikapi.Config{
+				APIURL:    s.URL,
+				AuthEmail: dummyAuthEmail,
+				AuthToken: dummyAuthToken,
+				RetryCfg: utils.RetryConfig{
+					MaxAttempts:          tt.retryMax,
+					MinDelay:             durationPtr(1 * time.Microsecond),
+					MaxDelay:             durationPtr(10 * time.Microsecond),
+					RetryableStatusCodes: tt.retryStatusCodes,
+					RetryableMethods:     tt.retryMethods,
+				},
+			})
+
+			// act
+			result, err := c.Users.
+				Create(context.Background(), tt.user)
+
+			// assert
+			t.Logf("Got result: %v, err: %v", result, err)
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+
+			assert.Equal(t, len(h.responses), len(h.requests), "invalid number of requests")
+			for _, r := range h.requests {
+				assert.Equal(t, http.MethodPost, r.method)
+				assert.Equal(t, "/user", r.url_.Path)
+				assert.Equal(t, dummyAuthEmail, r.header.Get(authEmailKey))
+				assert.Equal(t, dummyAuthToken, r.header.Get(authAPITokenKey))
+				assert.Equal(t, tt.expectedRequestBody, testutil.UnmarshalJSONToIf(t, r.body))
+			}
+			assert.Equal(t, tt.expectedResult, result)
+
+		})
+	}
+}
+
+
+type spyHTTPHandler struct {
```

Comment:
Please de-duplicate this handler with the one in APIv6 tests. I suggest to move it to _kentikapi/internal/testutil/http.go_ and reuse in both API v6 and API v5 tests.
There is simpler spy handler there, so I suggest to rename this one to e.g. MultipleResponseSpyHTTPHandler

Reply:
OK, it will work anyway.